src/main/java/uk/gov/companieshouse/service/CompanyOfficerService.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.companieshouse.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.ApiClientService;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.officers.CompanyOfficerApi;
+import uk.gov.companieshouse.exception.ServiceException;
+import uk.gov.companieshouse.mapper.CompanyOfficerMapper;
+import uk.gov.companieshouse.model.db.dissolution.DissolutionDirector;
+import uk.gov.companieshouse.repository.DissolutionRepository;
+
+import java.util.List;
+
+@Service
+public class CompanyOfficerService {
+    private static final UriTemplate GET_OFFICERS_URI = new UriTemplate("/company/{companyNumber}/officers");
+
+    private ApiClientService apiClientService;
+
+    private CompanyOfficerMapper companyOfficerMapper;
+
+    private final DissolutionRepository repository;
+
+    @Autowired
+    public CompanyOfficerService(
+            ApiClientService apiClientService,
+            DissolutionRepository repository,
+            CompanyOfficerMapper companyOfficerMapper
+    ) {
+        this.apiClientService = apiClientService;
+        this.repository = repository;
+        this.companyOfficerMapper = companyOfficerMapper;
+    }
+
+    public List<CompanyOfficerApi> getCompanyOfficers(String companyNumber) throws ServiceException {
```

Comment:
feels like this class has too many responsibilities - one for fetching company officers and one for checking has enough directors been selected. Feels like we should have CompanyOfficersClient service that does all the HTTP work i.e. generating URLs, checking status codes etc and the service would just call this client
```
final  List<CompanyOfficerApi> officers = client.getCompanyOfficers(companyNumber);
```